### ISSUE TITLE
Minion hanging on initial highstate after upgrade 2015.5.3 => 2015.8.1

### ISSUE THREAD

**Issue by syphernl** [saltstack/salt#28292]
After upgrading Salt-minion from 2015.5.3 => 2015.8.1 on an Ubuntu 14.04 box it resulted in a hanging salt-minion highstate.

We use the scheduler to run a highstate every 15 minutes, if the minion is restarted the counter is reset to 0 and executes a highstate. 
However, on all tested systems the process simply hangs. 

Either in a very early stage:
```
root@saas:~# salt-call state.highstate
local:
    Data failed to compile:
----------
    The function "state.highstate" is running as PID 25687 and was started at 2015, Oct 26 12:01:22.281533 with jid 20151026120122281533
root@saas:~# strace -p 25687
Process 25687 attached
restart_syscall(<... resuming interrupted call ...>^CProcess 25687 detached
 <detached ...>
```
or right after reading the minion key.

I killed the process after waiting for ~10 minutes without anything happening.
If I then run a new `salt-call state.highstate` it works just fine and keeps on working fine.

**Comment by peter-slovak** [comment]
If you use any custom (own) modules or runners that in turn use salt.client.Caller(), take a look at #28283 and the proposed fix in #28302 .

**Comment by Ch3LL** [comment]
@syphernl thanks for the report.

**Comment by syphernl** [comment]
@peter-slovak We indeed have a few runners and returners which make use of `salt.client.Caller`. 

I hope the fix in PR #28302 (which has been merged already) is soon released & packaged since this is currently blocking the upgrade of our Salt environment.

**Comment by cachedout** [comment]
@syphernl We're aiming to have this fix included in 2015.8.2 which should be released in three weeks.
